### PR TITLE
Adding Multi-Graph Random

### PR DESCRIPTION
**Pull request by yoloyi**
Adding Multi-Graph Random
===
* 增加多图随机，避免图片单一
* 增加用户设置title image

### CODE REVIEW

File: src/asset.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    protected getRandomOne(images: string[]): string {
+        var n = Math.floor(Math.random() * images.length + 1) - 1;
+        return images[n];
+    }
+
+    protected getConfigImages(): string[] {
+        let configImages: string[] = Utility.getConfiguration().get<string[]>('configImages', [
+            "http://b-ssl.duitang.com/uploads/item/201806/04/20180604090459_gqqjo.jpg",
```
Comment (formulahendry) [inline src/asset.ts]: Is it needed to have default value?

File: src/asset.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        }
+        return image;
+    }
+
+    public getTitle(): string {
+        return Utility.getConfiguration().get<string>('showTitle', '小哥哥，小哥哥，代码写久了，该休息啦~');
+    }
+
+    protected getRandomOne(images: string[]): string {
+        var n = Math.floor(Math.random() * images.length + 1) - 1;
```
Comment (formulahendry) [inline src/asset.ts]: const n

File: src/asset.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            "http://img5.imgtn.bdimg.com/it/u=2196122296,3201462689&fm=26&gp=0.jpg"
+        ]);
+        return configImages;
+    }
+
+    protected getDefaultImages(): string[] {
+        const dirPath = path.join(this.context.extensionPath, 'images');
+        let files: string[] = [];
+        const result = fs.readdirSync(dirPath);
+        result.forEach(function (item, index) {
```
Comment (formulahendry) [inline src/asset.ts]: how about putting all ycy images into a dedicated folder? Then no need to hard-code the PNGs.

File: src/asset.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    public getTitle(): string {
+        return Utility.getConfiguration().get<string>('showTitle', '小哥哥，小哥哥，代码写久了，该休息啦~');
+    }
+
+    protected getRandomOne(images: string[]): string {
+        var n = Math.floor(Math.random() * images.length + 1) - 1;
+        return images[n];
+    }
+
+    protected getConfigImages(): string[] {
```
Comment (formulahendry) [inline src/asset.ts]: A 'nice-to-have' feature: support image file in local folder

File: README.md [hunk 5]
```diff
@@ -17,3 +17,11 @@
 ## 配置
 
 * `ycy.reminderViewIntervalInMinutes`: 展示提醒页面的时间间隔（分钟）。 (默认值为**60**)
+* `ycy.showTitle`: 提示文字。 (默认值为**小哥哥，小哥哥，代码写久了，该休息啦~**)
```
Comment (formulahendry) [inline README.md]: `ycy.title`?

Comment (yoloyi) [reply README.md]: Yes , Title is better than showtitle

File: src/asset.ts (hunk 4 above)
Comment (yoloyi) [reply src/asset.ts]: cool idea . I am modifying it now.

File: README.md [hunk 6]
```diff
@@ -17,3 +17,11 @@
...
+
+* `ycy.isUseConfigImage`: 是否使用自己配置图片 (默认值为**false**)
+* `ycy.configImages`: 配置图片数组（需要搭配ycy.isUseConfigImage为true） (默认值为**默认图片**)
```
Comment (formulahendry) [inline README.md]: `ycy.customImages`?

File: src/asset.ts (hunk 3 above)
Comment (yoloyi) [reply src/asset.ts]: Cannot load resource outside of protocol root, so Can’t set the path  by the user。

File: src/asset.ts (hunk 4 above)
Comment (yoloyi) [reply src/asset.ts]: I tried it, Cannot load resource outside of protocol root,  so Can’t set the path by the custom。

**Comment by yoloyi** [comment]
> Overall are great! Just leave some comments.

I modified some code, please review again.

File: src/asset.ts (hunk 3 above)
Comment (yoloyi) [reply src/asset.ts]: I don't understand，new folder for image?  for example image/ycy?

Comment (formulahendry) [reply src/asset.ts]: Yes, so that you do not need `if (item === 'usage.png' || item === 'vote-qr.png')` anymore.

File: README.md [hunk 7]
```diff
@@ -16,12 +16,16 @@
...
-
-* `ycy.isUseConfigImage`: 是否使用自己配置图片 (默认值为**false**)
-* `ycy.configImages`: 配置图片数组（需要搭配ycy.isUseConfigImage为true） (默认值为**默认图片**)
-例如: [
-    "http://b-ssl.duitang.com/uploads/item/201806/04/20180604090459_gqqjo.jpg",
-    "http://img5.imgtn.bdimg.com/it/u=2196122296,3201462689&fm=26&gp=0.jpg"
+* `ycy.reminderViewIntervalInMinutes`: 展示提醒页面的时间间隔（分钟）。(默认值为**60**)
+* `ycy.title`: 提示文字。 (默认值为**小哥哥，小哥哥，代码写久了，该休息啦~**)
+
+* `ycy.type`: 类型：defalut:默认图；url：图片地址。(默认值为**defalut**)
```
Comment (formulahendry) [inline README.md]: Is `ycy.type:  defalut (默认图)；url (图片地址)` better?

File: package.json [hunk 8]
```diff
@@ -47,23 +47,20 @@
...
                 },
-                "ycy.configImages": {
+                "ycy.customImages": {
                     "type": "array",
-                    "default": [
-                        "http://b-ssl.duitang.com/uploads/item/201806/04/20180604090459_gqqjo.jpg",
-                        "http://img5.imgtn.bdimg.com/it/u=2196122296,3201462689&fm=26&gp=0.jpg"
-                    ],
-                    "description": "用户设置图片"
+                    "default": [],
```
Comment (formulahendry) [inline package.json]: I think we could put some default images here as reference:
```
 "http://b-ssl.duitang.com/uploads/item/201806/04/20180604090459_gqqjo.jpg",
"http://img5.imgtn.bdimg.com/it/u=2196122296,3201462689&fm=26&gp=0.jpg"
```

File: src/asset.ts [hunk 9]
```diff
@@ -4,53 +4,80 @@ import * as fs from 'fs';
...
     }
 
     public getImageUrl(): vscode.Uri | string {
-        let image: vscode.Uri | string = '';
-        if (this.isUseConfigImage()) {
-            image = this.getRandomOne(this.getConfigImages());
+        const type: string = this.getConfigType();
+        let images: vscode.Uri[] | string[];
+
+        // if (type === this.TYPE_LOCAL) {
```
Comment (formulahendry) [inline src/asset.ts]: Please remove all unused code~ Thanks!

File: src/asset.ts [hunk 10]
```diff
@@ -4,53 +4,80 @@ import * as fs from 'fs';
...
+        // } else
+        if (type === this.TYPE_URL_IMAGE) {
+            images = this.getCustomImages();
         } else {
-            image = vscode.Uri.file(this.getRandomOne(this.getDefaultImages())).with({ scheme: 'vscode-resource' });
+            images = this.getDefaultImages();
         }
-        return image;
-    }
+        // use forget setting customImages,get defalut images
```
Comment (formulahendry) [inline src/asset.ts]: - use -> user
- Add a space after `,`

File: src/asset.ts [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+export default class Asset {
+    // public readonly TYPE_LOCAL = 'local';
+    public readonly TYPE_URL_IMAGE = 'url';
+    public readonly TYPE_DEFAULT = 'defalut';
+
+    public constructor(private context: vscode.ExtensionContext) {
+    }
+
+    public getImageUrl(): vscode.Uri | string {
```
Comment (formulahendry) [inline src/asset.ts]: getImageUrl -> getImageUri

File: README.md (hunk 7 above)
Comment (yoloyi) [reply README.md]: it is scalability. if vscode supports  outside folder,  use type better

File: src/asset.ts (hunk 11 above)
Comment (yoloyi) [reply src/asset.ts]: Done~

File: src/asset.ts (hunk 10 above)
Comment (yoloyi) [reply src/asset.ts]: Done

File: src/asset.ts (hunk 9 above)
Comment (yoloyi) [reply src/asset.ts]: Done~

File: package.json (hunk 8 above)
Comment (yoloyi) [reply package.json]: Done~

File: src/asset.ts (hunk 3 above)
Comment (yoloyi) [reply src/asset.ts]: Done~

**Comment by yoloyi** [comment]
thanks review ! I modified Changes requested, please review again

**Comment by formulahendry** [comment]
@yoloyi THanks!
Yes, .vscode folder needs to be put into source control

**Comment by yoloyi** [comment]
Thanks

File: README.md [hunk 12]
```diff
@@ -16,4 +16,24 @@
 
 ## 配置
 
-* `ycy.reminderViewIntervalInMinutes`: 展示提醒页面的时间间隔（分钟）。 (默认值为**60**)
+* `ycy.reminderViewIntervalInMinutes`: 展示提醒页面的时间间隔（分钟）。(默认值为**60**)
+* `ycy.title`: 提示文字。 (默认值为**小哥哥，小哥哥，代码写久了，该休息啦~**)
+
+* `ycy.type`: 类型：defalut:默认图；url：图片地址。(默认值为**defalut**)
```
Comment (liyishuai) [inline README.md]: Is there a specific reason to use "defa**lu**t" rather than "defa**ul**t", or is it a typo?

**Pull request merged**